docs/gettinginvolved/functional-testing.rst

```diff
@@ -134,7 +134,49 @@ To run the functional tests, run the following command:
 OpenStack
 =========
 
-The configuration stanza for an private OpenStack deployment is the same as Rackspace above, but ``auth_plugin`` should be included, which refers to an authentication plugin provided by ``python-keystoneclient``.
+The configuration stanza for a private OpenStack deployment is similar to Rackspace (above), with a few notable differences:
+
+* ``auth_plugin`` should be included, which refers to an authentication plugin provided by ``python-keystoneclient``, and
+* ``provider: "openstack"`` should be included, if the top level key is not ``openstack``.
 
 If required, you may need to add additional fields.
 For more information, see :ref:`openstack-dataset-backend`.
+
+Devstack
+--------
+
+It is assumed that you have a working ``Devstack`` environment.
+Refer to document "Setting up a devstack instance" on Google Drive.
+
+To run the Cinder functional tests on ``devstack``:
+
+* boot a supported guest operating system in ``devstack``,
+* log into the guest and clone your branch of the Flocker source code and
+* install the Flocker dependencies in a ``virtualenv``.
+* Create ``$HOME/acceptance.yml`` containing:
+
+.. code:: yaml
+
+   # It is important to use ``devstack-openstack`` as the top-level name
```

Comment:
It indicates bad code, yes.

In `f.n.a.tests.blockdevicefactory`, `devstack-openstack` becomes the `cloud_provider` and each `cloud_provider` is  hardcoded in `MINIMUM_ALLOCATABLE_SIZES`.
Unfortunately the label `openstack` has been assigned `REALISTIC_BLOCKDEVICE_SIZE` which is a Rackspace limit.
It seems to me that the simplest thing would be to make it a configuration option,but I don't want to do that here.